xfields/beam_elements/beambeam3d.py

```diff
@@ -225,6 +272,14 @@ def __init__(self,
             assert (len(slices_other_beam_zeta_center_star)
                         == len(slices_other_beam_num_particles))
 
+        # beamstrahlung
+        if slices_other_beam_zeta_bin_width_star is not None:
+            assert not np.isscalar(slices_other_beam_zeta_bin_width_star), (
+                            'slices_other_beam_zeta_bin_width_star must be an array')
+            assert (len(slices_other_beam_zeta_bin_width_star)
+                        == len(slices_other_beam_num_particles))
```

Comment:
I made this array similar to the `slices_other_beam_zeta_center`, so that the user is required to input something. This is only used when the `flag_beamstrahlung` is on.

Reply:
Can lines 272-273 one 290-295 be merged in a single if block?

Reply:
merged it into one block